clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -764,18 +767,29 @@ static std::string computeUniqueSYCLXOCCName(StringRef Name,
     Result.push_back(c);
   }
 
+  if (!Result.empty() && (Result.front() == '_' || isDigit(Result.front()))) {
+    Result.front() = 'k';
+    ForceHash = true;
+  }
+
   /// The name alone is guaranteed to be unique, so if fits in the size, it is
   /// enough.
-  if (Result.size() < MaxXOCCSize)
+  if (Result.size() < MaxXOCCSize && !ForceHash)
     return Result;
 
   /// 9 for 8 characters of hash and an '_'.
   Result.erase(0, Result.size() - (MaxXOCCSize - 9));
```

Comment:
An easy way to simplify the following would be to suppress '_' from the hash string (62 characters among the 64) and start putting the hash at position 1 instead of 0. 
Because it is not impossible that the hash will be `________________` too...